src/TorchSharp/Utils/tensorboard/Utils.cs

```diff
@@ -0,0 +1,125 @@
+                    /// <param name="ncols"></param>
+                    /// <returns></returns>
+                    public static Tensor make_grid(Tensor I, long ncols = 8)
+                    {
+                        if (I.shape[1] == 1)
+                            I = cat(new Tensor[] { I, I, I }, 1);
+                        Trace.Assert(I.ndim == 4 && I.shape[1] == 3);
+                        long nimg = I.shape[0];
+                        long H = I.shape[2];
+                        long W = I.shape[3];
+                        ncols = Math.Min(ncols, nimg);
+                        long nrows = (long)Math.Ceiling((double)nimg / (double)ncols);
+                        Tensor canvas = zeros(new long[] { 3, H * nrows, W * ncols }, I.dtype);
+                        long i = 0;
+                        for (long r = 0; r < nrows; r++) {
+                            for (long c = 0; c < ncols; c++) {
+                                if (i >= nimg)
+                                    break;
+                                canvas.narrow(1, r * H, H).narrow(2, c * W, W).copy_(I[i]);
+                                i++;
+                            }
+                        }
+                        return canvas;
+                    }
+
+                    /// <summary>
+                    /// https://github.com/pytorch/pytorch/blob/6c30dc6ceed5542351b3be4f8043b28020f93f3a/torch/utils/tensorboard/_utils.py#L95
+                    /// </summary>
+                    /// <param name="tensor"> Image data </param>
+                    /// <param name="input_format"> Image data format specification of the form NCHW, NHWC, CHW, HWC, HW, WH, etc. </param>
+                    /// <returns></returns>
+                    public static Tensor convert_to_HWC(Tensor tensor, string input_format)
+                    {
+                        Trace.Assert(tensor.shape.Length == input_format.Length, $"size of input tensor and input format are different. tensor shape: ({string.Join(", ", tensor.shape)}), input_format: {input_format}");
+                        input_format = input_format.ToUpper();
+
+                        if (input_format.Length == 4) {
+                            long[] index = "NCHW".Select(c => Convert.ToInt64(input_format.IndexOf(c))).ToArray();
+                            Tensor tensor_NCHW = tensor.permute(index);
+                            Tensor tensor_CHW = make_grid(tensor_NCHW);
+                            return tensor_CHW.permute(1, 2, 0);
+                        }
+
+                        if (input_format.Length == 3) {
+                            long[] index = "HWC".Select(c => Convert.ToInt64(input_format.IndexOf(c))).ToArray();
+                            Tensor tensor_HWC = tensor.permute(index);
+                            if (tensor_HWC.shape[2] == 1)
+                                tensor_HWC = cat(new Tensor[] { tensor_HWC, tensor_HWC, tensor_HWC }, 2);
```

Comment:
Would it be faster/cheaper to use `Tensor.expand()` instead of `torch.cat()`

Reply:
🤔
```
>>> x = th.randn(3, 2, 3)
>>> x
tensor([[[ 1.0326,  0.7434, -0.8414],
         [-0.0487,  0.0576,  0.7714]],

        [[-0.4790,  1.1467,  1.6371],
         [-0.0265,  1.2140, -1.5797]],

        [[-1.0268,  0.4371,  1.6730],
         [ 0.6132, -0.9748, -0.4214]]])
>>> c = th.cat((x, x, x), 2)
>>> c.shape
torch.Size([3, 2, 9])
>>> c
tensor([[[ 1.0326,  0.7434, -0.8414,  1.0326,  0.7434, -0.8414,  1.0326,
           0.7434, -0.8414],
         [-0.0487,  0.0576,  0.7714, -0.0487,  0.0576,  0.7714, -0.0487,
           0.0576,  0.7714]],

        [[-0.4790,  1.1467,  1.6371, -0.4790,  1.1467,  1.6371, -0.4790,
           1.1467,  1.6371],
         [-0.0265,  1.2140, -1.5797, -0.0265,  1.2140, -1.5797, -0.0265,
           1.2140, -1.5797]],

        [[-1.0268,  0.4371,  1.6730, -1.0268,  0.4371,  1.6730, -1.0268,
           0.4371,  1.6730],
         [ 0.6132, -0.9748, -0.4214,  0.6132, -0.9748, -0.4214,  0.6132,
          -0.9748, -0.4214]]])
>>> x.expand(-1, -1, 9)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: The expanded size of the tensor (9) must match the existing size (3) at non-singleton dimension 2.  Target sizes: [-1, -1, 9].  Tensor sizes: [3, 2, 3]
```

Reply:
<img width="286" alt="image" src="https://user-images.githubusercontent.com/32535494/223122921-356e5d85-e2c5-445d-99b9-f20d049b0042.png">

Reply:
Done
https://github.com/dotnet/TorchSharp/pull/936/commits/97fd763384e5aaff2e5a89fb0e5e02a0aefbb807